File: backend/app/schemas/source_intelligence.py
import uuid

from pydantic import BaseModel, Field

from app.schemas.insights import InsightResponse

class SourceSummaryMetric(BaseModel):
    source_type: str
    label: str
    opportunities: int
    applications: int
    responses: int
    interviews: int
    response_rate: float | None
    interview_rate: float | None
    average_compass_score: float | None
    recruiter_contacts: int
    compensation_aligned: int
    basis: str


class SourceIntelligenceResponse(BaseModel):
    generated_at: str
    workspace_id: uuid.UUID | None = None
    summaries: list[SourceSummaryMetric] = Field(default_factory=list)
    insights: list[InsightResponse] = Field(default_factory=list)
    insufficient_data: list[str] = Field(default_factory=list)

File: frontend/src/types/sourceIntelligence.ts
import type { Insight } from "./insights";

export interface SourceSummaryMetric {
  source_type: string;
  label: string;
  opportunities: number;
  applications: number;
  responses: number;
  interviews: number;
  response_rate: number | null;
  interview_rate: number | null;
  average_compass_score: number | null;
  recruiter_contacts: number;
  compensation_aligned: number;
  basis: string;
}

export interface SourceIntelligenceResponse {
  generated_at: string;
  workspace_id: string | null;
  summaries: SourceSummaryMetric[];
  insights: Insight[];
  insufficient_data: string[];
}
